clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -4218,32 +4218,44 @@ def MemChrOp : CIR_Op<"libc.memchr"> {
 }
 
 //===----------------------------------------------------------------------===//
-// ReturnAddrOp
+// FuncAddrBuiltinOp
 //===----------------------------------------------------------------------===//
 
-def ReturnAddrOp : CIR_Op<"return_address"> {
-  let arguments = (ins UInt32:$level);
-  let summary = "The return address of the current function, or of one of its callers";
+def ReturnAddr : I32EnumAttrCase<"return_address", 1>;
+def FrameAddr : I32EnumAttrCase<"frame_address", 2>;
+def FuncAddrKind
```

Comment:
Thanks for trying to do some reuse here. I think these two operations should actually be on their own `cir.return_address` and `cir.frame_address`. You could still do some good reuse by factoring out common bits in a tablegen class.

Reply:
> Thanks for trying to do some reuse here. I think these two operations should actually be on their own `cir.return_address` and `cir.frame_address`. You could still do some good reuse by factoring out common bits in a tablegen class.

no problem, will make them separate Op.